src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs

```diff
@@ -618,97 +629,70 @@ private void ProcessMemberElements(object instance, XamlMemberDefinition member,
+
+					// Case 1: a single res-dict
+					if (member.Objects is [var singleChild] &&
+						TypeResolver.FindType(singleChild.Type)?.IsAssignableTo(typeof(ResourceDictionary)) == true)
 					{
-						if (innerDictionary.Members.FirstOrDefault(m => m.Member.Name == "ThemeDictionaries") is { } themeDictionaries)
+						if (propertyInfo.SetMethod == null)
 						{
-							foreach (var themeDictionary in themeDictionaries.Objects)
-							{
-								targetDictionary.ThemeDictionaries.Add(
-									GetResourceKey(themeDictionary),
-									LoadObject(themeDictionary, rootInstance));
-							}
+							throw new InvalidOperationException($"The property {propertyInfo} does not provide a setter (Line {member.LineNumber}:{member.LinePosition}");
 						}
 
-						if (innerDictionary.Members.FirstOrDefault(m => m.Member.Name == "MergedDictionaries") is { } mergedDictionaries)
+						var rd = (ResourceDictionary)LoadObject(singleChild, rootInstance)!;
+						if (IsFrameworkElementResources(propertyInfo))
 						{
-							foreach (var mergedDictionary in mergedDictionaries.Objects)
-							{
-								var newInstance = LoadObject(mergedDictionary, rootInstance);
-								if (newInstance is ResourceDictionary instanceAsDictionary)
-								{
-									targetDictionary.MergedDictionaries.Add(instanceAsDictionary);
-								}
-								else
-								{
-									throw new InvalidOperationException($"An object of type {newInstance?.GetType()} is not supported on MergedDictionaries");
-								}
-							}
+							((FrameworkElement)instance).Resources = rd;
 						}
-
-						if (innerDictionary.Members.FirstOrDefault(m => m.Member.Name == "_UnknownContent") is { } unknownContent)
+						else
 						{
-							dictionaryObjects = unknownContent.Objects;
+							propertyInfo.SetMethod.Invoke(instance, new[] { rd });
 						}
 					}
-
-					List<IDependencyObjectStoreProvider> delayResolutionList = new();
-					foreach (var child in dictionaryObjects)
+					// Case 2: zero-to-many non-res-dict resources
+					else if (member.Objects.All(x => TypeResolver.FindType(x.Type)?.IsAssignableTo(typeof(ResourceDictionary)) != true))
 					{
-						var item = LoadObject(child, rootInstance: rootInstance);
-
-						var resourceKey = GetResourceKey(child);
-						var resourceTargetType = GetResourceTargetType(child);
-
-						if (
-							item?.GetType() == typeof(Style)
-							&& resourceTargetType == null
-						)
+						if (propertyInfo.GetMethod == null)
 						{
-							throw new InvalidOperationException($"No target type was specified (Line {member.LineNumber}:{member.LinePosition}");
+							throw new InvalidOperationException($"The property {propertyInfo} does not provide a getter (Line {member.LineNumber}:{member.LinePosition}");
 						}
 
-						if ((resourceKey ?? resourceTargetType) is { } key)
+						var rd = (ResourceDictionary)propertyInfo.GetMethod.Invoke(instance, null)!;
```

Comment:
I may be missing something somewhere else but wouldn't this give me `null` if I had something like:

```cs
public class MyResDict : ResourceDictionary
{
	public ResourceDictionary MyInnerDict { get; set; }
}
```

with this xaml

```xml
<MyResDict>
	<MyResDict.MyInnerDict>
		<Color x:Key="SomeColorResource">Red</Color>
	</MyResDict.MyInnerDict>
</MyResDict>
```

unless I made it `public ResourceDictionary MyInnerDict { get; set; } = new()`?

Reply:
Oh nevermind I see :)

Reply:
this is also a failing condition in winui. so i guess it is fine to leave it like this.